flow-nodes/api-builder-plugin-md5/src/actions.js

```diff
@@ -1,18 +1,38 @@
 const crypto = require('crypto');
-
-function digest(req, outputs) {
-	const { data } = req.params;
+/**
+ * Action method.
+ *
+ * @param {object} params - A map of all the parameters passed from the flow.
+ * @param {object} options - The additional options provided from the flow
+ *	 engine.
+ * @param {object} options.pluginConfig - The service configuration for this
+ *	 plugin from API Builder config.pluginConfig['api-builder-plugin-pluginName']
+ * @param {object} options.logger - The API Builder logger which can be used
+ *	 to log messages to the console. When run in unit-tests, the messages are
+ *	 not logged.  If you wish to test logging, you will need to create a
+ *	 mocked logger (e.g. using `simple-mock`) and override in
+ *	 `MockRuntime.loadPlugin`.  For more information about the logger, see:
+ *	 https://docs.axway.com/bundle/API_Builder_4x_allOS_en/page/logging.html
+ * @param {*} [options.pluginContext] - The data provided by passing the
+ *	 context to `sdk.load(file, actions, { pluginContext })` in `getPlugin`
+ *	 in `index.js`.
+ * @return {*} The response value (resolves to "next" output, or if the method
+ *	 does not define "next", the first defined output).
+ */
+function digest(params, { logger }) {
+	const { data } = params;
 	if (data === undefined || data === null) {
-		return outputs.error(null, new Error('invalid argument: data'));
+		logger.error('invalid argument: data');
+		throw new Error('invalid argument: data');
 	}
 	try {
 		const hash = crypto
 			.createHash('md5')
 			.update(data)
 			.digest('hex');
-		outputs.next(null, hash);
+			return hash;
 	} catch (ex) {
-		outputs.error(null, ex);
+		throw new Error(ex);
```

Comment:
This try/catch is actually unnecessary now.  Any exception is now caught by the SDK action wrapper and will trigger the error output.  Watch for this kind of code - catch should be used to _handle_ the exception, but this doesn't actually handle it.  It just re-throws the error (wrapping it in another Error).  So, when you see a catch with a re-throw like this, you should be thinking "why are they re-throwing the exception?"

Reply:
Removed the try catch and I can remove this if in any other plugins as well.